crates/amaru-ledger/src/store.rs

```diff
@@ -67,6 +68,9 @@ pub enum StoreError {
 // ----------------------------------------------------------------------------
 
 pub trait ReadOnlyStore {
+    /// Get the current protocol parameters for a given epoch, or most recent one
+    fn get_protocol_parameters_for(&self, epoch: &Epoch) -> Result<ProtocolParameters, StoreError>;
```

Comment:
Not sure about this API; time will tell but I don't foresee we'll ever need to lookup protocol parameters from previous epochs.